### PR TITLE
Add GitHub Actions, remove AppVeyor

### PR DESCRIPTION
**Pull request by MSP-Greg**
Some of the changes are from dev-appveyor branch.

See https://github.com/MSP-Greg/skippy/actions/runs/3275428680

Note that Actions will not run when added from a PR...

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -6,13 +6,9 @@
 require 'minitest/autorun'
 require "minitest/reporters"
 require 'pathname'
-require 'pry'
+# require 'pry' # TODO: What was this used for?
```
Comment (thomthom) [inline test/test_helper.rb:9]: I think I was having trouble getting cucumber and debugging working from VSCode. And ended up using pry to inspect things... though, memory fails me...

File: test/library_source_test.rb [hunk 2]
```diff
@@ -91,6 +91,7 @@ def test_it_understand_local_absolute_git_paths
   end
 
   def test_it_understand_local_git_paths_with_backslashes
+    skip unless IS_WINDOWS
```
Comment (thomthom) [inline test/library_source_test.rb:94]: These failed outside of Windows?

**Comment by thomthom** [comment]
> It seems like SortedSet is a convenience, it's not used in any 'hot paths' (called thousands of times a second, etc), so I removed it. 

Looks like SortedSet is still being used? Did you add it back in again?

**Comment by thomthom** [comment]
> Not sure whether the test suite checks its use, so not sure about another PR for it...

I'm not sure myself, I'd have to double check.

File: test/library_source_test.rb (hunk 2 above)
Comment (MSP-Greg) [reply test/library_source_test.rb:94]: Yes.  Why would one have a path with backslashes outside of Windows?

I think most people expect Ruby to translate from slash to backslash when using Windows, I not sure if it's needed anymore.  I don't think people expect the converse (backslash to slash)?

Comment (thomthom) [reply test/library_source_test.rb:94]: I have to look closer at this. This didn't use to fail.

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-11, macos-12, windows-2019, windows-2022 ]
+        ruby: [ 2.5, 2.7, 3.1 ]
+        include:
+          - { os: windows-2022 , ruby: ucrt  }
+        exclude:
+          # keep macos jobs to 5
+          - { os: macos-11 , ruby: 3.1 }
```
Comment (thomthom) [inline .github/workflows/tests.yaml:35]: This will not run Ruby 3.1 on macOS 11?

**Comment by thomthom** [comment]
Fixed by #51

Thank you for taking the time to look into this!

**Pull request closed without merging**